packages/rsocket-websocket-client/src/WebsocketDuplexConnection.ts

```diff
@@ -15,9 +15,9 @@ export class WebsocketDuplexConnection
   constructor(private websocket: WebSocket) {
     super();
 
-    websocket.addEventListener("close", this.handleClosed);
-    websocket.addEventListener("error", this.handleError);
-    websocket.addEventListener("message", this.handleMessage);
+    websocket.addEventListener("close", this.handleClosed.bind(this));
+    websocket.addEventListener("error", this.handleError.bind(this));
+    websocket.addEventListener("message", this.handleMessage.bind(this));
```

Comment:
Have to bind the method reference when passing as an argument, otherwise, the `this` is bound to the `websocket` reference when the passed method reference is invoked.